src/NodaTime/Globalization/NodaFormatInfo.cs

```diff
@@ -398,9 +448,11 @@ internal static NodaFormatInfo GetFormatInfo(CultureInfo cultureInfo)
                 return InvariantInfo;
             }
             // Never cache (or consult the cache) for non-read-only cultures.
+            // We don't initialize eagerly in this case, to preserve previous behavior -
```

Comment:
I think this is the right decision (to not pre-initialize), but I'm not completely sure about the rationale here: does having a non-read-only `CultureInfo` input make it more likely that the output `NodaFormatInfo` will only be used briefly?